engine/database/output.go

```diff
@@ -0,0 +1,19 @@
+package database
+
+// Output represents an output of a transaction.
+// Fixme: This is not integrated with out db engine yet.
+type Output struct {
+	TxID       string  `gorm:"primaryKey"`
+	Vout       uint32  `gorm:"primaryKey"`
+	SpendingTX *string `gorm:"type:char(64)"`
+}
+
+// IsSpent returns true if the output is spent.
+func (o *Output) IsSpent() bool {
```

Comment:
This is in relation to suggestion to store data outputs also in this table.
Maybe this method should have name `IsUTXO` or `IsSpendable()` and be opposite of the `IsSpent`
Also to make it more fluent/readable in code, there could be a sibling method `IsNotUTXO()` or `IsNotSpendable()`

Reply:
I would add this during implementation of pay-by-paymail (PTPKH).
- because then I'll have both cases.